### PR TITLE
ESQL: Remove OPTIONS clause in FROM command

### PR DESCRIPTION
**Pull request by bpintea**
This remove the `OPTIONS` clause of the `FROM` command.

### CODE REVIEW

**Comment by bpintea** [comment]
Related: https://github.com/elastic/elasticsearch/pull/106636, https://github.com/elastic/elasticsearch/pull/107012, https://github.com/elastic/elasticsearch/pull/107013.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java [hunk 1]
```diff
@@ -793,24 +792,14 @@ static void writeDissect(PlanStreamOutput out, Dissect dissect) throws IOExcepti
     }
 
     static EsRelation readEsRelation(PlanStreamInput in) throws IOException {
-        Source source = in.readSource();
-        EsIndex esIndex = readEsIndex(in);
-        List<Attribute> attributes = readAttributes(in);
-        EsSourceOptions esSourceOptions = in.getTransportVersion().onOrAfter(TransportVersions.ESQL_ES_SOURCE_OPTIONS)
```
Comment (dnhatn) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java:799]: I think we need to keep serialization around (read/write but discard) to maintain bwc for users who do not use OPTIONS.

File: docs/reference/esql/esql-language.asciidoc [hunk 2]
```diff
@@ -20,7 +19,6 @@ include::esql-syntax.asciidoc[]
 include::esql-commands.asciidoc[]
 include::esql-functions-operators.asciidoc[]
 include::metadata-fields.asciidoc[]
-include::esql-index-options.asciidoc[]
```
Comment (marciw) [inline docs/reference/esql/esql-language.asciidoc:23]: Should we go ahead and remove the [actual file](https://github.com/elastic/elasticsearch/blob/2730fd390301d30bc3e7c521437872e755a3494e/docs/reference/esql/esql-index-options.asciidoc) too? 

(but it's also fine to do that in a later separate PR -- removing the `include` line does the trick)

**Review by marciw — APPROVED** [review]
Reviewed the docs changes -- LGTM!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java:799]: Thanks, indeed, we need to satisfy the serialisation requirements between `TransportVersions.ESQL_ES_SOURCE_OPTIONS` and "now" -- newly introduced `TransportVersions.ESQL_REMOVE_ES_SOURCE_OPTIONS`.

File: docs/reference/esql/esql-language.asciidoc (hunk 2 above)
Comment (bpintea) [reply docs/reference/esql/esql-language.asciidoc:23]: Thanks, removed.

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks @bpintea.

**Pull request merged**